### ISSUE TITLE
Package request: flexget

### ISSUE THREAD

**Issue by sgtsquiggs** [Entware-ng/Entware-ng#104]
http://flexget.com

Installing via easy_install or pip fails when trying to install the python regex module:

```
Installing collected packages: regex, rebulk, babelfish, guessit, futures, pytz, tzlocal, apscheduler, Werkzeug, itsdangerous, flask, aniso8601, flask-restful, ordereddict, flask-restplus, cherrypy, webassets, flask-assets, cssmin, flask-compress, flask-login, pyparsing, enum34, pathlib, pyScss, pytvmaze, flexget
  Running setup.py install for regex ... error
    Complete output from command /opt/bin/python2.7 -u -c "import setuptools, tokenize;__file__='/share/CACHEDEV1_DATA/.qpkg/Entware-ng/tmp/pip-build-qnyLLK/regex/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /opt/tmp/pip-wUaK8E-record/install-record.txt --single-version-externally-managed --compile:
    /opt/lib/python2.7/site-packages/setuptools/dist.py:284: UserWarning: Normalizing '2016.01.10' to '2016.1.10'
      normalized_version,
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-2.7
    copying Python2/regex.py -> build/lib.linux-x86_64-2.7
    copying Python2/_regex_core.py -> build/lib.linux-x86_64-2.7
    copying Python2/test_regex.py -> build/lib.linux-x86_64-2.7
    warning: build_py: byte-compiling is disabled, skipping.

    running build_ext
    building '_regex' extension
    creating build/temp.linux-x86_64-2.7
    creating build/temp.linux-x86_64-2.7/Python2
    x86_64-openwrt-linux-gnu-gcc -fno-strict-aliasing -O2 -pipe -fomit-frame-pointer -fno-caller-saves -fhonour-copts -Wno-error=unused-but-set-variable -DNDEBUG -fno-inline -DNDEBUG -O2 -pipe -fomit-frame-pointer -fno-caller-saves -fhonour-copts -Wno-error=unused-but-set-variable -fPIC -I/opt/include/python2.7 -c Python2/_regex.c -o build/temp.linux-x86_64-2.7/Python2/_regex.o
    Python2/_regex.c:46:20: fatal error: Python.h: No such file or directory
     #include "Python.h"
                        ^
    compilation terminated.
    error: command 'x86_64-openwrt-linux-gnu-gcc' failed with exit status 1

    ----------------------------------------
Command "/opt/bin/python2.7 -u -c "import setuptools, tokenize;__file__='/share/CACHEDEV1_DATA/.qpkg/Entware-ng/tmp/pip-build-qnyLLK/regex/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /opt/tmp/pip-wUaK8E-record/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /share/CACHEDEV1_DATA/.qpkg/Entware-ng/tmp/pip-build-qnyLLK/regex
```

I installed the gcc package but I couldn't find a python-dev package.

**Comment by kolyagora** [comment]
I use flexget installed via pip in Entware-ng on a QNAP x86_64 NAS without any problems. Having a package with a proper startup script would be nice though.

**Comment by kolyagora** [comment]
Just tried to upgrade with `pip install --upgrade flexget` and got the same error. It must be regex package related as It wasn't required before, at least not in v1.2.422 which I have installed.

**Comment by sgtsquiggs** [comment]
`pip install flexget==1.2.422` worked for me, thanks. I guess we really need to request a `python-regex` package? It seems to be the real problem.

**Comment by zyxmon** [comment]
python-regex is in the repo now.
There is also a way to compile python packages natively.
flexget requires a lot of python packages. Here is a list that is not full:
FeedParser-5.2.1 MarkupSafe-0.23 PyRSS2Gen-1.1 PyYAML-3.11 SQLAlchemy-1.0.11
Werkzeug-0.11.3 aniso8601-1.1.0 apscheduler-3.0.5 argparse-1.4.0 babelfish-0.5.5
beautifulsoup4-4.4.1 cherrypy-4.0.0 cssmin-0.2.0 enum34-1.1.2 flask-0.10.1 flask-assets-0.11
flask-compress-1.3.0 flask-login-0.3.2 flask-restful-0.3.5 flask-restplus-0.7.2
functools32-3.2.3.post2 futures-3.0.4 guessit-0.10.3 html5lib-0.9999999 itsdangerous-0.24
jinja2-2.8 jsonschema-2.5.1 ordereddict-1.1 path.py-8.1.2 pathlib-1.0.1 pbr-1.8.1
plumbum-1.6.1.post0 progressbar-2.3 pyScss-1.3.4 pynzb-0.1.0 pyparsing-2.0.7
python-dateutil-2.4.2 pytvmaze-1.4.4 pytz-2015.7 rpyc-3.3.0 six-1.10.0 stevedore-1.10.0
tmdb3-0.7.2 tzlocal-1.2 webassets-0.11.1
It is much easier to install flexget with pyton-pip that is now in the repo.

**Comment by sgtsquiggs** [comment]
@zyxmon, this installs fine with python-regex via `pip install flexget`. Thanks again.

**Comment by kolyagora** [comment]
@zyxmon, thank you for adding python-regex package!

> There is also a way to compile python packages natively.

As @sgtsquiggs already noticed, there is no python-dev package in entware-ng. Will it be enough to add python header files to gcc include path to build a module on a device?

**Comment by zyxmon** [comment]
Here is an instruction to compile on NAS (wiki)
https://github.com/Entware-ng/Entware-ng/wiki/Using-gcc-%28native-compilation%29
You MUST run `gcc_env.sh` before compiling from Python to set RPATH and other options!

**Comment by kolyagora** [comment]
@zyxmon, thanks for pointing out.

**Comment by lancethepants** [comment]
I've noticed when compiling some python modules that have c-code, it remembers the compiler that cross-compiled python, so it tries to use that.  In my case (my own project) I had something like arm-linux-gcc, so I have to symlink that to gcc when doing a native compile against a cross-compiled version of python.

**Comment by pedro-MD11** [comment]
I'm having issues with a fresh install of Entware-ng on my Asus RT-AC66U. pip install flexget fails giving a few warnings and a code error -11. Installing python-regex didn't work either.

`Collecting flexget
  Using cached FlexGet-1.2.452.tar.gz
    Complete output from command python setup.py egg_info:
    running egg_info
    creating pip-egg-info/FlexGet.egg-info
    writing requirements to pip-egg-info/FlexGet.egg-info/requires.txt
    writing pip-egg-info/FlexGet.egg-info/PKG-INFO
    writing top-level names to pip-egg-info/FlexGet.egg-info/top_level.txt
    writing dependency_links to pip-egg-info/FlexGet.egg-info/dependency_links.txt
    writing entry points to pip-egg-info/FlexGet.egg-info/entry_points.txt
    writing manifest file 'pip-egg-info/FlexGet.egg-info/SOURCES.txt'
    reading manifest file 'pip-egg-info/FlexGet.egg-info/SOURCES.txt'
    reading manifest template 'MANIFEST.in'
    warning: no previously-included files matching '*' found under directory 'tests'

    ----------------------------------------
Command "python setup.py egg_info" failed with error code -11 in /tmp/pip-build-1qakiI/flexget
`

**Comment by zyxmon** [comment]
pip needs a lot of space in your HOME dir. If HOME variable points to RAM - this can be an issue. Try to run something like

```
export HOME=<path to storage>
```

before running pip

**Comment by pedro-MD11** [comment]
I also think it is memory related somehow. I've tried using the --cache-dir and -b options from pip to point to my storage with random results. Tried with the export HOME as well, same thing.
I've also checked memory usage with free -m and it often doesn't drop below 100MB and my swap file is not being used at all.
Sometimes it manages to build the FlexGet package and FeedParser, so they are in cache. I couldn't get past SQLAlchemy.
I have installed flexget before on this router with the old entware going the long way to installing pip, but an upgrade broke it and I was trying to build fresh. I do have a backup of my previous installation just in case.
